nexus/reconfigurator/planning/src/blueprint_editor/sled_editor.rs

```diff
@@ -745,6 +791,189 @@ impl ActiveSledEditor {
+                            );
+                            info.status = NoopConvertSledStatus::Ineligible(
+                                MupdateOverride {
+                                    mupdate_override_id: inv_override,
+                                    zones,
+                                },
+                            );
+                        }
+                    }
+                }
+
+                // Set all zone image sources to InstallDataset. This is an
+                // acknowledgement of the current state of the world.
+                let zone_ids: Vec<_> = self
+                    .zones(BlueprintZoneDisposition::is_in_service)
+                    .map(|zone| (zone.id, zone.kind()))
+                    .collect();
+
+                let mut zones = IdOrdMap::with_capacity(zone_ids.len());
+                for (zone_id, kind) in zone_ids {
+                    let old_image_source = self.zones.set_zone_image_source(
+                        &zone_id,
+                        BlueprintZoneImageSource::InstallDataset,
+                    )?;
+                    let item = EnsureMupdateOverrideUpdatedZone {
+                        zone_id,
+                        kind,
+                        old_image_source,
+                        new_image_source:
+                            BlueprintZoneImageSource::InstallDataset,
+                    };
+                    zones.insert_unique(item).expect(
+                        "self.zones is a BTreeMap so zone IDs are unique",
+                    );
+                }
+
+                // Clear out the pending MGS update for this sled.
+                let prev_mgs_update = match pending_mgs_update {
+                    Entry::Vacant(_) => None,
+                    Entry::Occupied(entry) => Some(Box::new(entry.remove())),
+                };
+
+                // Clear out the host phase 2 information for this sled as well.
+                let prev_host_phase_2 = self.set_host_phase_2(
+                    BlueprintHostPhase2DesiredSlots::current_contents(),
+                );
+
+                Ok(EnsureMupdateOverrideAction::BpSetOverride {
+                    inv_override,
+                    prev_bp_override: bp_override,
+                    zones,
+                    prev_mgs_update,
+                    prev_host_phase_2,
+                })
+            }
+            (None, Some(bp_override)) => {
+                // The blueprint says there's an override in place, but the
+                // inventory doesn't. This means that the sled has removed its
+                // override that was set in the above branch.
+                //
+                // However, the blueprint's remove_mupdate_override remains in
+                // place until all zones' image sources can be noop-converted to
+                // Artifact. We do this to minimize the number of different
+                // versions of software that exist.
```

Comment:
Is this bit of the comment right? I think the `TODO` down on line 918 means it's not true yet.

I'm not sure this is where I'd implement this bit (see my earlier comment about adding a third check to whether it's okay to proceed to subsequent steps).